cmd/tealdbg/localLedger.go

```diff
@@ -0,0 +1,218 @@
+// Copyright (C) 2019-2020 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package main
+
+import (
+	"fmt"
+	"math/rand"
+
+	"github.com/algorand/go-algorand/config"
+	"github.com/algorand/go-algorand/crypto"
+	"github.com/algorand/go-algorand/data/basics"
+	"github.com/algorand/go-algorand/data/transactions"
+	"github.com/algorand/go-algorand/data/transactions/logic"
+	"github.com/algorand/go-algorand/ledger"
+)
+
+type balancesAdapter struct {
+	balances   map[basics.Address]basics.AccountData
+	txnGroup   []transactions.SignedTxn
+	groupIndex int
+	proto      config.ConsensusParams
+	round      int
+}
+
+func makeAppLedger(
+	balances map[basics.Address]basics.AccountData, txnGroup []transactions.SignedTxn,
+	groupIndex int, proto config.ConsensusParams, round int, latestTimestamp int64,
+	appIdx basics.AppIndex, painless bool,
+) (logic.LedgerForLogic, appState, error) {
+
+	if groupIndex >= len(txnGroup) {
+		return nil, appState{}, fmt.Errorf("invalid groupIndex %d exceed txn group length %d", groupIndex, len(txnGroup))
+	}
+	txn := txnGroup[groupIndex]
+
+	accounts := []basics.Address{txn.Txn.Sender}
+	for _, addr := range txn.Txn.Accounts {
+		accounts = append(accounts, addr)
+	}
```

Comment:
This is a very tiny nit, but I believe it should be possible to do
```
	accounts := []basics.Address{txn.Txn.Sender}
	accounts = append(accounts, txn.Txn.Accounts...)
```
instead here.

Reply:
done!